integration-tests/hibernate-reactive-panache/src/test/java/io/quarkus/it/panache/reactive/TransactionalUniAsserterTest.java

```diff
@@ -18,16 +20,21 @@ public class TransactionalUniAsserterTest {
     @Test
     public void testTransactionalUniAsserter(TransactionalUniAsserter asserter) {
         assertNotNull(asserter);
-        asserter.assertThat(() -> Panache.currentTransaction(), transaction -> {
+        asserter.assertThat(Panache::currentTransaction, transaction -> {
             assertNotNull(transaction);
             assertFalse(transaction.isMarkedForRollback());
             asserter.putData("tx", transaction);
         });
-        asserter.assertThat(() -> Panache.currentTransaction(), transaction -> {
+        asserter.assertThat(Panache::currentTransaction, transaction -> {
             assertNotNull(transaction);
             assertFalse(transaction.isMarkedForRollback());
             assertNotEquals(transaction, asserter.getData("tx"));
         });
     }
 
+    @Test
+    @RunOnVertxContext
+    public void testFailure(final TransactionalUniAsserter asserter) {
+        assertThrows(AssertionFailedError.class, asserter::fail);
```

Comment:
In fact, `asserter::fail` does not result in an exception. You would need to subscribe to the underlying `Uni`.

Unfortunately, JUnit5 does support the Junit4-style `@Test(expected = NullPointerException.class)`. So we can't easily assert the failure...?

Reply:
Ah, right. Thanks